File: messagebus/src/vespa/messagebus/testlib/simplereply.cpp
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include "simplereply.h"
#include "simpleprotocol.h"

namespace mbus {

SimpleReply::SimpleReply(const string &str) :
    Reply(),
    _value(str)
{ }

SimpleReply::~SimpleReply() { }

void
SimpleReply::setValue(const string &value)
{
    _value = value;
}

const string &
SimpleReply::getValue() const
{
    return _value;
}

const string &
SimpleReply::getProtocol() const
{
    return SimpleProtocol::NAME;
}

uint32_t
SimpleReply::getType() const
{
    return SimpleProtocol::REPLY;
}

} // namespace mbus

File: messagebus/src/main/java/com/yahoo/messagebus/test/SimpleReply.java
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.messagebus.test;

import com.yahoo.messagebus.Reply;
import com.yahoo.text.Utf8String;

/**
 * @author havardpe
 */
public class SimpleReply extends Reply {

    private String value;

    public SimpleReply(String value) {
        this.value = value;
    }

    public int getType() {
        return SimpleProtocol.REPLY;
    }

    public Utf8String getProtocol() {
        return SimpleProtocol.NAME;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }
}
